### PR TITLE
feat(flist): add support for FuseSoC flags

### PR DESCRIPTION
**Pull request by shareefj**
This commit updates the `flist` tool with support for passing FuseSoC flags on the command line just as you would when calling `fusesoc`.

Flags can be set using:

	flist --flag myflag <system>
	flist --flag +myflag <system>

or unset using

	flist --flag -myflag <system>

or you can use key=value pairs

	flist --flag sram=sim

which would create the flag `sram_sim`.

### CODE REVIEW

**Review by Jonah-Foley — CHANGES_REQUESTED** [review]
I tried this commit with the makefile in multicore-qc and it threw up the error:

```
Traceback (most recent call last):
  File "/home/jfoley/multicore-qc/control/.venv/bin/flist", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/jfoley/multicore-qc/control/.venv/lib64/python3.11/site-packages/edalize/tools/flist.py", line 294, in main
    filelist = flist(
               ^^^^^^
  File "/home/jfoley/multicore-qc/control/.venv/lib64/python3.11/site-packages/edalize/tools/flist.py", line 243, in flist
    src = list(glob_dir.glob("**/*.f"))[0]
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
```

It seems there are a few places where `build_root` needs to be replaced with `work_root` (im not sure how these two work together). For reference, this is how the simulation directory looks when you run the flist tool with `--work-root=work` (before it errors out)

```
├── fastlink.py
├── frame.py
├── Makefile
├── phaser_mvp_core.f
├── process.ipynb
├── __pycache__
│   ├── fastlink.cpython-311-pytest-8.1.1.pyc
│   ├── frame.cpython-311-pytest-8.1.1.pyc
│   ├── testbench.cpython-311-pytest-8.1.1.pyc
│   └── test_mvp.cpython-311-pytest-8.1.1.pyc
├── testbench.py
├── test_mvp.py
└── work
    ├── Makefile
    ├── nq_phaser_phaser_mvp_core_0.eda.yml
    ├── nq_phaser_phaser_mvp_core_0.f
    ├── phased_duc_lut.init
    └── src
        ├── misoc_cores_duc_0
        ├── misoc_cores_duc-verilog_netlist_0
        ├── nq_phaser_csr_0
        ├── nq_phaser_dac_data_0
        ├── nq_phaser_fastlink_0
        ├── nq_phaser_interpolator_0
        ├── nq_phaser_phaser_mvp_core_0
        ├── nq_phaser_spi_master_0
        ├── quartiq_phaser_checker_0
        ├── quartiq_phaser_checker-verilog_netlist_0
        ├── quartiq_phaser_interpolate_channel_0
        ├── quartiq_phaser_interpolate_channel-verilog_netlist_0
        ├── quartiq_phaser_sample_mux_0
        ├── quartiq_phaser_sample_mux-verilog_netlist_0
        ├── quartiq_phaser_unframer_0
        └── quartiq_phaser_unframer-verilog_netlist_0
```
may give you an idea of whats wrong.

**Comment by shareefj** [comment]
Can you provide the exact command line needed to replicate that error?  It works for me.

**Comment by Jonah-Foley** [comment]
From within `/multicore-qc/control/hw/boards/phaser/top_phaser_mvp/dv` running 
```
make 
```

Here is the full trace:


```
(.venv) [jfoley@localhost dv]$ ls
data  fastlink.py  frame.py  Makefile  process.ipynb  __pycache__  testbench.py  test_mvp.py
(.venv) [jfoley@localhost dv]$ make
rm -f results.xml
make -f Makefile results.xml
make[1]: Entering directory '/home/jfoley/multicore-qc/control/hw/boards/phaser/top_phaser_mvp/dv'
flist -o phaser_mvp_core.f --work-root work -f include_waivers phaser_mvp_core
INFO: user flag=include_waivers
INFO: Preparing nq:phaser:csr:0
INFO: Preparing nq:phaser:dac_data:0
INFO: Preparing nq:phaser:spi_master:0
INFO: Preparing nq:utils:generators:0
INFO: Preparing misoc:cores:duc:0
INFO: Preparing quartiq:phaser:checker:0
INFO: Preparing quartiq:phaser:interpolate_channel:0
INFO: Preparing quartiq:phaser:sample_mux:0
INFO: Preparing quartiq:phaser:unframer:0
INFO: Preparing nq:phaser:fastlink:0
INFO: Preparing nq:phaser:interpolator:0
INFO: Preparing nq:phaser:phaser_mvp_core:0
INFO: Generating misoc:cores:duc-verilog_netlist:0
INFO: Generating quartiq:phaser:checker-verilog_netlist:0
INFO: Generating quartiq:phaser:interpolate_channel-verilog_netlist:0
INFO: Generating quartiq:phaser:sample_mux-verilog_netlist:0
INFO: Generating quartiq:phaser:unframer-verilog_netlist:0
Traceback (most recent call last):
  File "/home/jfoley/multicore-qc/control/.venv/bin/flist", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/jfoley/multicore-qc/control/.venv/lib64/python3.11/site-packages/edalize/tools/flist.py", line 294, in main
    filelist = flist(
               ^^^^^^
  File "/home/jfoley/multicore-qc/control/.venv/lib64/python3.11/site-packages/edalize/tools/flist.py", line 243, in flist
    src = list(glob_dir.glob("**/*.f"))[0]
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
```

**Comment by shareefj** [comment]
Have you pip installed the latest edalize-plugins branch?

**Comment by Jonah-Foley** [comment]
Yes, confident in this fact else it would have failed out earlier with an unknown command line argument `--work-root`

File: src/edalize/tools/flist.py [hunk 1]
```diff
@@ -159,6 +162,7 @@ def flist(
 
     Arguments:
         name: VLNV of the core to process or just the N if it resolves to a unique name.
+        flags: an optional list of addition FuseSoC flags to apply
```
[the remark this answers is not in the document]
Comment (shareefj) [reply src/edalize/tools/flist.py:165]: Done.

**Review by Jonah-Foley — APPROVED** [review]
LGTM

**Pull request merged**